mdm/api/src/main/java/com/gentics/mesh/core/data/dao/BinaryDao.java

```diff
@@ -93,4 +100,49 @@ default Transactional<? extends HibBinary> create(String hash, long size, Binary
 	 * @return
 	 */
 	Transactional<Stream<HibBinary>> findAll();
+
+	/**
+	 * Get the manipulation variants of this binary image representation. On any other binary data returns null.
+	 * 
+	 * @return
+	 */
+	Result<? extends HibImageVariant> getVariants(HibBinary binary, InternalActionContext ac);
+
+	/**
+	 * Get the manipulation variants of this binary image representation. On any other binary data returns null.
+	 * 
+	 * @return
+	 */
+	HibImageVariant getVariant(HibBinary binary, ImageManipulation variant, InternalActionContext ac);
+
+	/**
+	 * Create the image manipulation variant,
+	 * 
+	 * @param variant
+	 * @return created variant
+	 */
+	HibImageVariant createVariant(HibBinary binary, ImageVariantRequest variant, InternalActionContext ac, boolean throwOnExisting);
+
+	/**
+	 * Delete the image manipulation variant,
+	 * 
+	 * @param variant
+	 */
+	void deleteVariant(HibBinary binary, ImageVariantRequest variant, InternalActionContext ac, boolean throwOnAbsent);
+
+	/**
+	 * Create the image manipulation variants, optionally deleting all other existing unused variants.
+	 * 
+	 * @param variant
+	 * @return image variants of the binary after the operation.
+	 */
+	Result<? extends HibImageVariant> createVariants(HibBinary binary, Collection<ImageVariantRequest> variants, InternalActionContext ac, boolean deleteOther);
+
+	/**
+	 * Delete the image manipulation variants. Depending on the 'deleteOther' flag, either the parameter variants or other ones are to be deleted.
+	 * 
+	 * @param variant
+	 * @return image variants of the binary after the operation.
+	 */
+	Result<? extends HibImageVariant> deleteVariants(HibBinary binary, Collection<ImageVariantRequest> variant, InternalActionContext ac, boolean deleteOther);
```

Comment:
Maybe it's clearer to make two methods out of this:
deleteVariants() which deletes the given variants
retainVariants() which deletes all but the the given variants